dragnet/compat.py

```diff
@@ -8,20 +8,259 @@
+    """
+    Converts any bytes-like items in input dict to string-like values, with respect to python version
+
+    Parameters
+    ----------
+    dict_ : dict
+        any bytes-like objects contained in the dict will be converted to a string (assumed utf-8 encoding)
+    include_keys : bool, default=True
+        if True, cast keys to a string, else ignore
+    include_values : bool, default=True
+        if True, cast values to a string, else ignore
+    kwargs:
+        encoding: str, default: 'utf-8'
+            encoding to be used when decoding bytes
+    """
+    old_keys = dict_.keys()
+    old_vals = dict_.values()
+    new_keys = str_list_cast(old_keys, **kwargs) if include_keys else old_keys
+    new_vals = str_list_cast(old_vals, **kwargs) if include_vals else old_vals
+    new_dict = dict(zip(new_keys, new_vals))
+    return new_dict
+
+
+def bytes_dict_cast(dict_, include_keys=True, include_vals=True, **kwargs):
+    """
+    Converts any string-like items in input dict to bytes-like values, with respect to python version
+
+    Parameters
+    ----------
+    dict_ : dict
+        any string-like objects contained in the dict will be converted to bytes (assumed utf-8 encoding)
+    include_keys : bool, default=True
+        if True, cast keys to bytes, else ignore
+    include_values : bool, default=True
+        if True, cast values to bytes, else ignore
+    kwargs:
+        encoding: str, default: 'utf-8'
+            encoding to be used when encoding string
+    """
+    old_keys = dict_.keys()
+    old_vals = dict_.values()
+    new_keys = bytes_list_cast(old_keys, **kwargs) if include_keys else old_keys
+    new_vals = bytes_list_cast(old_vals, **kwargs) if include_vals else old_vals
+    new_dict = dict(zip(new_keys, new_vals))
+    return new_dict
+
+
+def str_block_cast(block,
+                   include_text=True,
+                   include_link_tokens=True,
+                   include_css=True,
+                   include_features=True,
+                   **kwargs):
+    """
+    Converts any bytes-like items in input Block object to string-like values, with respect to python version
+
+    Parameters
+    ----------
+    block : blocks.Block
+        any bytes-like objects contained in the block object will be converted to a string (assumed utf-8 encoding)
+    include_text : bool, default=True
+        if True, cast text to a string, else ignore
+    include_link_tokens : bool, default=True
+        if True, cast link_tokens to a string, else ignore
+    include_css : bool, default=True
+        if True, cast css to a string, else ignore
+    include_features : bool, default=True
+        if True, cast features to a string, else ignore
+    kwargs:
+        encoding: str, default: 'utf-8'
+            encoding to be used when decoding bytes
+    """
+    block.text = str_cast(block.text, **kwargs)                    if include_text        else block.text
```

Comment:
Using whitespace like this is strongly discouraged in Python. Also, in the case that `include_*` is False, there's no need to assign the value back to itself, right? So maybe this would be both more readable and more performant:

```
if include_text:
    block.text = str_cast(block.text, **kwargs)
if include_link_tokens:
    block.link_tokens = str_list_cast(block.link_tokens, **kwargs)
...
```